documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+# Before you Start
+
+[TypeScript Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) is the overall design guideline of the client SDK.
+Make sure you are familiar with concepts such as "Service Client" and "Packaging".
+
+Make sure you are familiar with Git.
+
+See [Prerequisite](https://www.npmjs.com/package/@azure-tools/typespec-ts) of `@azure-tools/typespec-ts`. Make sure that NodeJS is installed.
+
+# Generate SDK
+
+## Use TypeSpec defined in REST API specifications
+
+:ballot_box_with_check: The package structure in the azure-rest-api-specs repository should follow [these guidelines](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md#structure).
+
+Here are the samples for\
+[Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)\
+[RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/Face/tspconfig.yaml)
+
+Please make sure `service-dir`, `package-dir`, `packageDetails`, `isModularLibrary`, `generateMetadata`, `flavor`(for typespec-ts) is correctly configured. `experimentalExtensibleEnums`, `enableOperationGroup`, `hierarchyClient` are the optional configs.
+If you want to generate samples with typespec-ts, you should add 
+```
+examples-directory: "{project-root}/examples"
+generateSample:true
+```
+in your tspconfig.yaml
+
+
+- "parameters.service-dir.default" would be `sdk/<service>`
+- "options.@azure-tools/typespec-ts.package-dir" would be `<module>`
+
+SDK module would be generated under the SDK project folder at `sdk/<service>/<module>`.
+
+### Generate Code with code-gen-pipeline tool
```

Comment:
@kazrael2119 could you refer to the [document for Go](https://github.com/Azure/azure-sdk-for-go/pull/23837/files) to add the corresponding content for JS?

Reply:
done

Reply:
@raych1  I think what ziwei authored is better, since we didn't make linux shell be required to customers.

Reply:
Some steps are missing here for JS:
- clone spec repo
- clone JS Sdk repo
- run `sh .scripts/automation_init.sh`